cockroachcloud/architecture.md

```diff
@@ -0,0 +1,76 @@
+
+### Performance 
+
+Serverless clusters scale based on your application's workload. Free clusters include 250M RUs with 10M RUs reserved for [burst performance](#concepts) and 5GiB of storage per month. They have a guaranteed baseline performance of 100 RUs per second, or up to 100 QPS. If you set a budget for your cluster, you will have access to additional resources without throttling.
+
+Depending on your workload, your budget will be used differently. For example, a cluster using very little storage space will have more of its budget available for Request Units, and vice versa. If you hit your budget, your cluster will be throttled down to baseline performance levels. If this occurs, you can opt to increase your budget, adjust your workload to stay within the current spend limit, or stay at the baseline performance level until the next month.
+
+Storage always gets first priority in the budget since you need to be able to store the data first and foremost. The remainder of the budget is allocated to burst and baseline performance. You can theoretically use up all of your budget for burst performance in the first few minutes of a cluster being created. If this happens, the cluster will be throttled back to the baseline performance and can reaccumulate burst capacity by using fewer RUs.
+
+Serverless clusters also have the ability to scale to zero and consume no resources when there are no active queries. When there are no active queries, you will pay for the storage your app is using, but not for Request Units. To avoid wasted resources, CockroachCloud automatically pauses free clusters that are inactive, which is defined by having no connection to the cluster for two consecutive minutes. Once the user attempts to reconnect to the cluster, the cluster will automatically resume. Pausing, resuming, and scaling clusters is a fully-managed process and will not disrupt or affect the user experience. However, it is important for your application to have connection retry logic in the event of node restarts or network disruptions. For more information, see the [Production Checklist](production-checklist.html).
+
+### Architecture
+
+Traffic comes in from the public internet and is routed by the cloud provider’s load balancer to a Kubernetes cluster that hosts CockroachDB. However, unlike a self-hosted CockroachDB cluster where the SQL and KV layers run in the same process, our serverless architecture fully decouples the SQL layer from the KV layer so that both layers run in separate processes running in separate Kubernetes pods.
```

Comment:
The average reader probably won't know what we're talking about here, because "SQL Layer" and "KV Layer" are not concepts that we have publicly documented. I'd do one of three things:

1. Remove mention of SQL and KV layers from this doc
2. Explain in this doc what the SQL and KV layers are
3. Include docs on the SQL and KV layers in our [Architecture Overview](https://www.cockroachlabs.com/docs/v21.1/architecture/overview.html). Basically I think the KV layer is just all of the layers that aren't the "SQL Layer."

Reply:
I think proxy/sql/kv layer info was requested, but will double check with Lakshmi. If I go with option 2, what do you think of adding an abridged version of [this explanation](https://docs.google.com/document/d/1_QRXGSrqQYktLwyJquv7NMuFut77zRqrS4UXiLA00_w/edit#heading=h.q3oshpx8alwt)?

Reply:
update on this: going to leave sql and kv out for now and add info as a follow-up pr